code/datums/components/seethrough_mob.dm

```diff
@@ -0,0 +1,124 @@
+/datum/component/seethrough_mob
+	///The atom that enables our dark magic
+	var/obj/render_source_atom
+	///The fake version of ourselves
+	var/image/trickery_image
+	///Which alpha do we animate towards?
+	var/target_alpha
+	///How long our faze in/out takes
+	var/animation_time
+	///Does this object let clicks from players its transparent to pass through it
+	var/clickthrough
+	///Is the seethrough effect currently active
+	var/is_active
+
+/datum/component/seethrough_mob/Initialize(target_alpha = 100, animation_time = 0.5 SECONDS, clickthrough = TRUE)
+	. = ..()
+
+	if(!ismob(parent))
+		return COMPONENT_INCOMPATIBLE
+
+	src.target_alpha = target_alpha
+	src.animation_time = animation_time
+	src.clickthrough = clickthrough
+	src.is_active = FALSE
+	var/datum/action/toggle_seethrough/action = new(src)
+	action.Grant(parent)
+
+///Set up everything we need to trick the client and keep it looking normal for everyone else
+/datum/component/seethrough_mob/proc/trick_mob()
+	SIGNAL_HANDLER
+
+	var/mob/fool = parent
+	var/datum/hud/our_hud = fool.hud_used
+	for(var/atom/movable/screen/plane_master/seethrough in our_hud.get_true_plane_masters(SEETHROUGH_PLANE))
+		seethrough.unhide_plane(fool)
+
+	render_source_atom = new()
+
+	var/static/uid = 0
+	uid++
+	fool.render_target = "*transparent_bigmob[uid]"
+	fool.vis_contents.Add(render_source_atom)
+	render_source_atom.render_source = fool.render_target
+
+	render_source_atom.appearance_flags |= ( RESET_COLOR | RESET_TRANSFORM)
+
+	render_source_atom.vis_flags |= (VIS_INHERIT_ID | VIS_INHERIT_PLANE | VIS_INHERIT_LAYER)
+
+	var/icon/current_mob_icon = icon(fool.icon, fool.icon_state)
+	render_source_atom.pixel_x = -fool.pixel_x
+	render_source_atom.pixel_y = ((current_mob_icon.Height() - 32) * 0.5)
+
+	trickery_image = new(render_source_atom)
+	trickery_image.loc = render_source_atom
+	trickery_image.override = TRUE
+
+	trickery_image.pixel_x = 0
+	trickery_image.pixel_y = 0
+
+	if(clickthrough)
+		//Special plane so we can click through the overlay
+		SET_PLANE_EXPLICIT(trickery_image, SEETHROUGH_PLANE, fool)
+
+	fool.client.images += trickery_image
+
+	animate(trickery_image, alpha = target_alpha, time = animation_time)
+
+	RegisterSignal(fool, COMSIG_MOB_LOGOUT, PROC_REF(on_client_disconnect))
+
+///Remove the screen object and make us appear solid to ourselves again
+/datum/component/seethrough_mob/proc/untrick_mob()
+	var/mob/fool = parent
+	animate(trickery_image, alpha = 255, time = animation_time)
+	UnregisterSignal(fool, COMSIG_MOB_LOGOUT)
+
+	//after playing the fade-in animation, remove the image and the trick atom
+	addtimer(CALLBACK(src, PROC_REF(clear_image), trickery_image, fool.client), animation_time)
+
+///Remove the image and the trick atom
+/datum/component/seethrough_mob/proc/clear_image(image/removee, client/remove_from)
+	var/atom/atom_parent = parent
+	atom_parent.appearance_flags &= ~KEEP_TOGETHER
+	atom_parent.render_target = null
+	qdel(render_source_atom)
```

Comment:
ditto

Reply:
Can you rephrase that?

Reply:
Whoops! Just make sure `render_source_atom` is deleted when the component is also deleted.